apps/server/src/modules/tool/service/external-tool.service.ts

```diff
@@ -44,6 +39,29 @@ export class ExternalToolService {
 		return created;
 	}
 
+	async updateExternalTool(toUpdate: ExternalToolDO): Promise<ExternalToolDO> {
+		if (toUpdate.config instanceof Oauth2ToolConfigDO) {
+			const toUpdateOauthClient: ProviderOauthClient = this.mapper.mapDoToProviderOauthClient(
+				toUpdate.name,
+				toUpdate.config
+			);
+			const loadedOauthClient: ProviderOauthClient = await this.oauthProviderService.getOAuth2Client(
```

Comment:
should this fail if no client is found? then you might be able to skip the error handling below

Reply:
Yes, it should fail, because we dont want to have diverging data